src/js/webSerial.js

```diff
@@ -30,12 +30,22 @@ class WebSerial extends EventTarget {
 
         this.logHead = "SERIAL: ";
 
+        this.port_counter = 0;
+        this.ports = null;
         this.port = null;
         this.reader = null;
         this.writer = null;
         this.reading = false;
 
         this.connect = this.connect.bind(this);
+
+        navigator.serial.addEventListener("connect", device => {
+            this.dispatchEvent(new CustomEvent("addedDevice", { detail: device.target }));
+        });
+
+        navigator.serial.addEventListener("disconnect", device => {
+            this.dispatchEvent(new CustomEvent("removedDevice", { detail: device.target }));
+        });
```

Comment:
Wondering if we need this here? We are not using the `event.detail` later on and all this is doing is proxying the events.

Could we just use `navigator.serial` directly in the port_handler under the `isWeb` branch?

Reply:
I think is good to "isolate" the webserial. If we have a webserial.js to isolate all the serial operations, then it's responsible of indicating all the important events, like sending datan, new devices, etc. 
I think more, it will be good to send events like "connected", "disconnected", etc. Will make a lot more simple the rest of the code, but maybe for other PR.
About the event detail, I will remove it, now we don't use it, but maybe we can act different in a future depending on the type of device connected or similar. My idea was to "add" or "remove" the device from the list in the port handler without calling the "getPorts" again, but the code is not ready for that, at least while we maintain compatibility with nw.js.